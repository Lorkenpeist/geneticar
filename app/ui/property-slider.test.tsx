import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import PropertySlider from "./property-slider";

describe(PropertySlider.name, () => {
  const discreteSlider = (
    <PropertySlider
      id="discrete-slider"
      name="discreteSlider"
      label="Discrete Slider"
      min={1}
      max={5}
      step={1}
      defaultValue={3}
    />
  );
  const continuousSlider = (
    <PropertySlider
      id="continuous-slider"
      name="continuousSlider"
      label="Continuous Slider"
      min={1}
      max={5}
      defaultValue={3}
    />
  );

  it("renders discrete slider correctly", () => {
    const tree = render(discreteSlider);
    expect(tree.container).toMatchSnapshot();
  });

  it("renders continuous slider correctly", () => {
    const tree = render(continuousSlider);
    expect(tree.container).toMatchSnapshot();
  });

  it.each([
    { slider: discreteSlider, newValue: 4 },
    { slider: continuousSlider, newValue: 4 },
    { slider: continuousSlider, newValue: 3.14 },
  ])("updates its value when dragged", async ({ slider, newValue }) => {
    const tree = render(slider, { container: document.createElement("form") });

    const sliderInput = tree.getByRole("slider");

    fireEvent.change(sliderInput, { target: { value: newValue } });

    // Confirm that the display text is in sync with the slider
    expect(tree.getByText(new RegExp(slider.props.label))).toHaveTextContent(
      new RegExp(`^${slider.props.label}: ${newValue}$`),
    );
    // Confirm that the form will be submitted with the correct value
    expect(tree.container).toHaveFormValues({
      [slider.props.name]: `${newValue}`,
    });
  });
});
