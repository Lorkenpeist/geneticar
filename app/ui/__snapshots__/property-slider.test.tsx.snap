// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertySlider renders continuous slider correctly 1`] = `
<div>
  <div
    class="flex flex-col items-center text-xs font-medium"
  >
    <label
      for="continuous-slider"
    >
      Continuous Slider
      : 
      3
    </label>
    <div
      class="flex flex-row gap-1 items-center"
    >
      <div
        class="w-5 text-right"
      >
        1
      </div>
      <input
        id="continuous-slider"
        max="5"
        min="1"
        name="continuousSlider"
        step="0.01"
        type="range"
        value="3"
      />
      <div
        class="w-7 text-left"
      >
        5
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertySlider renders discrete slider correctly 1`] = `
<div>
  <div
    class="flex flex-col items-center text-xs font-medium"
  >
    <label
      for="discrete-slider"
    >
      Discrete Slider
      : 
      3
    </label>
    <div
      class="flex flex-row gap-1 items-center"
    >
      <div
        class="w-5 text-right"
      >
        1
      </div>
      <input
        id="discrete-slider"
        list="discrete-slider:markers"
        max="5"
        min="1"
        name="discreteSlider"
        step="1"
        type="range"
        value="3"
      />
      <datalist
        id="discrete-slider:markers"
      >
        <option
          value="1"
        />
        <option
          value="2"
        />
        <option
          value="3"
        />
        <option
          value="4"
        />
        <option
          value="5"
        />
      </datalist>
      <div
        class="w-7 text-left"
      >
        5
      </div>
    </div>
  </div>
</div>
`;
